src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -118,12 +189,205 @@
     return rankings;
   }
 
-  // parse the given file into a List of CastVoteRecords for tabulation
+  // helper to extract the required CVRContest from the input CVR
+  CVRContest getCvrContest(CVR cvr, Contest contestToTabulate) {
+    CVRContest cvrContestToTabulate = null;
+    // find current snapshot
+    CVRSnapshot currentSnapshot = null;
+    if (cvr.CVRSnapshot != null) {
+      for (CVRSnapshot cvrSnapshot : cvr.CVRSnapshot) {
+        if (cvrSnapshot.ObjectId.equals(cvr.CurrentSnapshotId)) {
+          currentSnapshot = cvrSnapshot;
+          break;
+        }
+      }
+    }
+    // find CVRContest which matches the Contest we are tabulating
+    if (currentSnapshot != null && currentSnapshot.CVRContest != null) {
+      for (CVRContest cvrContest : currentSnapshot.CVRContest) {
+        if (cvrContest.ContestId.equals(contestToTabulate.ObjectId)) {
+          cvrContestToTabulate = cvrContest;
+          break;
+        }
+      }
+    }
+    return cvrContestToTabulate;
+  }
+
+  void parseXml(List<CastVoteRecord> castVoteRecords) {
+    try {
+
+      // load XML
+      XmlMapper xmlMapper = new XmlMapper();
+      xmlMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      FileInputStream inputStream = new FileInputStream(new File(filePath));
+      CastVoteRecordReport cvrReport = xmlMapper.readValue(inputStream, CastVoteRecordReport.class);
+
+      // Find the contest for tabulation
+      Contest contestToTabulate = null;
+      for (Election election : cvrReport.Election) {
+        for (Contest contest : election.Contest) {
+          if (contest.Name.equals(this.contestId)) {
+            contestToTabulate = contest;
+            break;
+          }
+        }
+      }
+
+      if (contestToTabulate == null) {
+        Logger.severe("Contest \"%s\" from config file not found!", this.contestId);
+        throw new CvrParseException();
+      }
+
+      // build a map of Candidates
+      HashMap<String, Candidate> candidateById = new HashMap<>();
+      for (Election election : cvrReport.Election) {
+        for (Candidate candidate : election.Candidate) {
+          candidateById.put(candidate.ObjectId, candidate);
+        }
+      }
+
+      // build a map of ContestSelections
+      HashMap<String, ContestSelection> contestSelectionById = new HashMap<>();
+      for (ContestSelection contestSelection : contestToTabulate.ContestSelection) {
+        if (contestSelection.CandidateIds.length > 1) {
+          Logger.warning(
+              "Multiple CandidateIds found in ContestSelection \"%s\". "
+                  + " Only the first one will be processed.",
+              contestSelection.ObjectId);
+        }
```

Comment:
I saw your note below about not supporting this yet. So should this be a fatal error instead of just a warning?